physics/cires_vert_orodis_v1.F90

```diff
@@ -0,0 +1,1026 @@
+       tauhcr  = coef_h2 *hcrit*hcrit
+
+       IF (fr <= fcrit_gfs) then
+         tauogw  = taulin   
+         tautot  = taulin   
+         taulee  =  0.
+         drlee(:) = 0.
+       ELSE                      !fr > fcrit_gfs
+         tauogw  = tauhcr
+         taulee  = max(tautot - tauogw, 0.0)
+         if (taulee > 0.0 ) hdsw = sqrt(taulee/coef_h2) 
+!  approximate drlee(k) between  [izb, klwb]
+! find klwb and decrease taulee(izb) => taulee(klwb) = 0.
+! above izb  tau
+         if (mkz2 > mkz2min.and. krefj > 2 .and. taulee > 0.0) then
+
+           mkz = sqrt(mkz2)
+           call mflow_tauz(levs, up, vp, tp, qp, dp,   zpm, zpi, &
+                pmid, pint, rho, ui, vi, ti,  bn2i, bvi,  rhoi)
+          
+           kph  = max(izb, 2)
+           phmkz(:) = 0.0
+           klwb = max(izb, 1)
+           kotr = levs+1
+           phmkz(kph-1) = fr  ! initial Phase of the Lee-OGW
+
+           loop_lwb_gfs18:  do k=kph, levs
+
+             umd(k) = max(ui(k)*xn +vi(k)*yn, velmin)
+             umd2   =(coro- umd(k))*(coro- umd(k))
+             umd2   = max(umd2, velmin*velmin)
+             mkz2   = bn2i(k)/umd2 - kx2
+             if ( mkz2 > mkz2min ) then
+               mkz      = sqrt(mkz2)
+               frlocal  = max(hdsw*bvi(k)/umd(k), frlwb) 
+               phmkz(k) = phmkz(k-1) + mkz*(zpm(k)-zpm(k-1))
+               if( ( phmkz(k) >= zbr_pi ) .and. (frlocal > frlin)) klwb = k
+             else 
+               kotr = k
+               exit loop_lwb_gfs18
+             endif
+           enddo   loop_lwb_gfs18
+!
+!
+           do k=kph, kotr-1
+
+             if (klwb > kph .and. k < klwb) then
+               drlee(k)   = -(tautot -tauogw)/(zpi(kph) - zpi(klwb))
+               tau_src(k) = tautot + (zpi(k) - zpi(klwb))*drlee(k)
+               drlee(k)   = drlee(k)/rho(k)
+             else if ( k >= klwb .and. k < kotr) then
+               tau_src(k) = tauogw
+               drlee(k)   = 0.0
+             endif
+           enddo  
+           kdswj = klwb   ! assign to the "low-level" wave breaking
+         endif              !  mkz2 > mkz2min.and. krefj > 2 .and. taulee > 0.0
+       ENDIF               !fr > fcrit_gfs
+
+
+     ENDIF     !strbase='gfs2017' .or. strbase='kd_2005'		 
+ 
+
+! output : taulin,  taufrb, taulee, xn, yn, umag, kxw/kxridge	
+!       print *,  krefj, levs,  tauogw,  tautot , ' ugwp_taub_oro ' 
+!
+      end subroutine ugwp_taub_oro
+!
+!--------------------------------------
+! 
+!     call ugwp_oro_lsatdis( krefj, levs,  tauogw(j),  tautot(j), tau_src, kxw,      &
+!           fcor(j), c2f2(j), up, vp, tp, qp, dp, zpm, zpi, pmid1, pint1,            &
+!	   xn, yn, umag, drtau, kdis_oro)
+ 
+      subroutine ugwp_oro_lsatdis( krefj, levs,  tauogw,  tautot,  tau_src,     &
+           kxw, fcor,  kxridge, up, vp, tp, qp, dp, zpm, zpi, pmid, pint,       &
+           xn, yn, umag, drtau, kdis)
+
+      use ugwp_common_v1,       only : bnv2min, grav, pi, pi2, dw2min, velmin, rgrav
```

Comment:
Constants should come from the host model through the argument list, if possible.

Reply:
Done.